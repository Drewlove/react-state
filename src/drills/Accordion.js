import React, {Component} from 'react'; 

class Accordion extends Component{
    state = {
        currentSectionIndex: null
    }

    handleClick = (i)=> {
        this.setState({
            currentSectionIndex: i
        })
    }
    
    renderList(){
        return this.props.section.map((item, i ) => {
            return <li key={i}>
            <button  onClick={()=>this.handleClick(i)}>{item.title}</button>
            <p>{this.state.currentSectionIndex ===  i ? 
                this.props.section[i].content 
                : null}
            </p>
            </li>
            }
        )}

    render(){
        return (
            <div>
                <ul>
                    {!!this.props.section && this.renderList()}
                </ul>
            </div>     
        )
    }
}

export default Accordion 