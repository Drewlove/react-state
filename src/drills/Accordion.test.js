import React from 'react'; 
import ReactDOM from 'react-dom'; 
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme'; 
import toJson from 'enzyme-to-json'
import Accordion from "./Accordion"; 
import { finished } from 'stream';

const sectionsProp = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ]

it('renders without errors', ()=> {
    const div = document.createElement('div'); 
    ReactDOM.render(<Accordion />, div); 
    ReactDOM.unmountComponentAtNode(div)
})

it('renders without props', ()=> {
    const wrapper = shallow(<Accordion />)
    expect(toJson(wrapper)).toMatchSnapshot()
})

it('renders when user clicks section', ()=> {
    const wrapper = shallow(<Accordion section={sectionsProp}/>)
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
})

it('only renders last section clicked', ()=> {
    const wrapper = shallow(<Accordion section={sectionsProp} />)
    wrapper.find('button').at(0).simulate('click')
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
})














//
// it('renders empty given no tabs', () => {
//     const wrapper = shallow(<Tabs />)
//     expect(toJson(wrapper)).toMatchSnapshot()
//   })

//   it('renders the first tab by default', () => {
//     const wrapper = shallow(<Tabs tabs={tabsProp} />)
//     expect(toJson(wrapper)).toMatchSnapshot()
//   })

//   it('closes the first tab and opens any clicked tab', ()=> {
//       const wrapper = shallow(<Tabs tabs={tabsProp}/>)
//       wrapper.find('button').at(1).simulate('click')
//       expect(toJson(wrapper)).toMatchSnapshot()
//   })