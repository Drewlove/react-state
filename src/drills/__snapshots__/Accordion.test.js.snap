// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`only renders last section clicked 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        onClick={[Function]}
      >
        Section 1
      </button>
      <p />
    </li>
    <li
      key="1"
    >
      <button
        onClick={[Function]}
      >
        Section 2
      </button>
      <p>
        Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
      </p>
    </li>
    <li
      key="2"
    >
      <button
        onClick={[Function]}
      >
        Section 3
      </button>
      <p />
    </li>
  </ul>
</div>
`;

exports[`renders when user clicks section 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        onClick={[Function]}
      >
        Section 1
      </button>
      <p />
    </li>
    <li
      key="1"
    >
      <button
        onClick={[Function]}
      >
        Section 2
      </button>
      <p>
        Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
      </p>
    </li>
    <li
      key="2"
    >
      <button
        onClick={[Function]}
      >
        Section 3
      </button>
      <p />
    </li>
  </ul>
</div>
`;

exports[`renders without props 1`] = `
<div>
  <ul />
</div>
`;
